nano/node/blockprocessor.cpp

```diff
@@ -241,6 +242,7 @@ void nano::block_processor::process_batch (std::unique_lock<std::mutex> & lock_a
 		}
 	}
 	lock_a.unlock ();
+	auto write_guard = write_database_queue.wait (nano::writer::process_batch);
```

Comment:
None of the `auto write_guard` variables in the PR appears to be used - can they be removed?

Reply:
It is similar to something like `lock_guard`, which isn't directly called but will pop the writer off the front of the queue when it goes out of scope.

Reply:
Ah yes, missed that in the surface scan.

Could we name it `scoped_write_guard` ?